.ai-doc/20-dependencies/react-i18next.md

```diff
@@ -0,0 +1,714 @@
+
+## 🔗 References
+
+- [Manager React Shell Client](./manager-react-shell-client.md)
+- [Manager React Core Application](./manager-react-core-application.md)
+- [Common Translations](../10-architecture/common-translations.md)
+- [React i18next Documentation](https://react.i18next.com/)
+- [i18next Core Documentation](https://www.i18next.com/)
+
+## ⚡ React i18next v16 Changes
+
+**Major Update:** react-i18next v16 requires **React 18+** for TypeScript users. For JavaScript users, v16 is functionally equivalent to v15.1.4.
+
+**Key v16 Features:**
+- Enhanced React Compiler and React.memo compatibility with i18n wrapper
+- Improved Trans component stability and rendering
+- Better TypeScript type definitions
+- Fixed issues with special characters in Trans component
+
+## 📘 Guidelines / Implementation
+
+### Package Installation
+
+```json
+{
+  "dependencies": {
+    "react-i18next": "^16.2.4",
+    "i18next": "^23.8.2",
+    "i18next-http-backend": "^2.4.3"
+  }
+}
+```
+
+### Basic i18n Setup
+
+#### i18n Configuration
+
+```typescript
+// i18n.ts
+import i18n from 'i18next';
+import { initReactI18next } from 'react-i18next';
+import Backend from 'i18next-http-backend';
+import { initI18n } from '@ovh-ux/manager-react-core-application';
+
+// Initialize i18n with shell integration
+export const initializeI18n = async (locale: string, availableLocales: string[]) => {
+  const t = await initI18n(locale, availableLocales);
+  return t;
+};
+
+// Or configure manually
+i18n
+  .use(Backend)
+  .use(initReactI18next)
+  .init({
+    lng: 'fr_FR',
+    fallbackLng: 'fr_FR',
+    debug: process.env.NODE_ENV === 'development',
+    
+    backend: {
+      loadPath: '/translations/{{lng}}/{{ns}}.json',
+    },
+    
+    ns: ['common', 'dashboard', 'onboarding'],
+    defaultNS: 'common',
+    
+    interpolation: {
+      escapeValue: false, // React already escapes values
+    },
+    
+    react: {
+      useSuspense: false,
+    },
+  });
+
+export default i18n;
+```
+
+#### Shell Integration
+
+```typescript
+// i18n.ts
+import { initI18n } from '@ovh-ux/manager-react-core-application';
+import { useContext } from 'react';
+import { ShellContext } from '@ovh-ux/manager-react-shell-client';
+
+export const useI18n = () => {
+  const { shell } = useContext(ShellContext);
+  
+  const initializeI18n = async () => {
+    if (shell) {
+      const locale = await shell.i18n.getLocale();
+      const availableLocales = await shell.i18n.getAvailableLocales();
+      
+      return await initI18n(locale, availableLocales.map(l => l.key));
+    }
+  };
+  
+  return { initializeI18n };
+};
+```
+
+### Translation Files Structure
+
+#### File Organization
+
+```
+public/translations/
+├── fr_FR/
+│   ├── common.json
+│   ├── dashboard.json
+│   └── onboarding.json
+├── en_GB/
+│   ├── common.json
+│   ├── dashboard.json
+│   └── onboarding.json
+└── de_DE/
+    ├── common.json
+    ├── dashboard.json
+    └── onboarding.json
```

Comment:
Its the other way round we use in manager. We create namespaces as folder and each folder contains the translation files for all locales.